src/main/resources/web/public/peer.dwrtc.js

```diff
@@ -0,0 +1,195 @@
+"use strict"
+
+class EventDispatcher {
+  constructor() {
+    this.events = []
+  }
+
+  on(event, callback) {
+    const handlers = this.events[event] || []
+    handlers.push(callback)
+    this.events[event] = handlers
+  }
+
+  trigger(event, data) {
+    const handlers = this.events[event] || []
+    handlers.forEach(handler => handler(data))
+  }
+}
+
+class SignalingMessage {
+  constructor(recipientSessionId, messageBody) {
+    this.type = "SignalingMessage"
+    this.recipientSessionId = recipientSessionId
+    this.messageBody = messageBody
+  }
+}
+
+class DWRTC {
+  constructor(isInitiator, initialPeerId, webSocketUrl) {
+    this.isInitiator = isInitiator
+    if (this.isInitiator) this.otherPeerId = initialPeerId
+    this.webSocketUrl = webSocketUrl
+    this.dispatcher = new EventDispatcher()
+
+    console.log(
+      `Started DWRTC with isInitiator: ${isInitiator}, initialPeerId: ${initialPeerId}`
+    )
+  }
+
+  /**
+   * Sets the class up
+   * This is a separate method to the constructor, since constructors are not allowed to be async
+   */
+  async setup() {
+    await this.setupSocket()
+    await this.startSimplePeer()
+  }
+
+  /**
+   * Get video / audio stream
+   */
+  getStream(video = true, audio = true) {
+    let stream
+    try {
+      stream = navigator.mediaDevices.getUserMedia({
+        video: video,
+        audio: audio
+      })
+    } catch (error) {
+      throw error
+    }
+    return stream
+  }
+
+  /**
+   * Initialize the peer with the information we have so far
+   */
+  async startSimplePeer() {
+    const stream = await this.getStream()
+
+    this.peer = new window.SimplePeer({
+      initiator: this.isInitiator,
+      stream: stream
+    })
+
+    this.peer.on("signal", data => {
+      // Peer wants to send signaling data
+      console.debug(`Send Signal message: ${data}`)
```

Comment:
```suggestion
      console.debug(`Send Signal message: ${JSON.stringify(data)}`)
```

This is so we can debug the sent message